### PR TITLE
Reorganize various functions and files

### PR DESCRIPTION
**Pull request by VoR0220**
* Move maker into chains package
* Delete writers in `util` package and move them to respective packages (cm_files, cm_files_test, cm_archives to maker/package related writers go to jobs)
* Delete prerun from pkgs do command, move commands that were previously there into RunPackage function
* Move various loader functions from maker into the loaders package. 
* Rids go-wire as a dependency.

### CODE REVIEW

File: loaders/chains.go [hunk 1]
```diff
@@ -14,6 +15,48 @@ import (
...
+	var typ = definitions.BlankChainType()
+	var chainType = viper.New()
+
+	if err := getSetup(fileName, chainType); err != nil {
+		return nil, err
+	}
+
+	// marshall file
+	if err := chainType.Unmarshal(typ); err != nil {
+		return nil, fmt.Errorf("\nSorry, the marmots could not figure that chain types file out.\nPlease check your chain type definition file is properly formatted.\nERROR =>\t\t\t%v", err)
```
Comment (zramsay) [inline loaders/chains.go]: ahem

File: loaders/chains.go [hunk 2]
```diff
@@ -14,6 +15,48 @@ import (
...
+	extName := filepath.Ext(file)
+	bName := file[:len(file)-len(extName)]
+
+	cfg.AddConfigPath(path)
+	cfg.SetConfigName(bName)
+	cfg.SetConfigType(strings.Replace(extName, ".", "", 1))
+
+	// load file
+	if err := cfg.ReadInConfig(); err != nil {
+		return fmt.Errorf("\nSorry, the marmots were unable to load the file: (%s). Please check your path.\nERROR =>\t\t\t%v", fileName, err)
```
Comment (zramsay) [inline loaders/chains.go]: ibid

File: pkgs/jobs/jobs_transact.go [hunk 3]
```diff
@@ -113,7 +113,7 @@ func RegisterNameJob(name *definitions.RegisterName, do *definitions.Do) (string
 
 			n := fmt.Sprintf("%s:%s", record[0], record[1])
 			// TODO: fix this... simple and naive result just now.
-			if err = util.WriteJobResultCSV(n, r); err != nil {
+			if err = WriteJobResultCSV(n, r); err != nil {
 				log.Error("ERROR =>")
```
Comment (zramsay) [inline pkgs/jobs/jobs_transact.go]: this is ugly too

File: pkgs/jobs/writers.go [hunk 4]
```diff
@@ -8,20 +8,13 @@ import (
 	"path/filepath"
 	"reflect"
 
-	"github.com/eris-ltd/eris-cli/definitions"
-	"github.com/eris-ltd/eris-cli/log"
-
 	// MARMOT: dependency on go-wire needs to be removed.
 	"github.com/tendermint/go-wire"
```
Comment (zramsay) [inline pkgs/jobs/writers.go]: kill it

File: pkgs/run_package.go [hunk 5]
```diff
@@ -5,12 +5,19 @@ import (
...
 	"github.com/eris-ltd/eris-cli/util"
 )
 
 func RunPackage(do *definitions.Do) error {
+
+	// clears job_outputs file
+	jobs.ClearJobResults()
+	// useful for debugging
+	PrintPathPackage(do)
+
```
Comment (zramsay) [inline pkgs/run_package.go]: should be unexported

File: loaders/chains.go (hunk 1 above)
Comment (VoR0220) [inline loaders/chains.go]: This can be gotten rid of in a separate PR.

File: pkgs/jobs/writers.go (hunk 4 above)
Comment (VoR0220) [inline pkgs/jobs/writers.go]: Separate PR

File: pkgs/run_package.go (hunk 5 above)
Comment (VoR0220) [inline pkgs/run_package.go]: Agreed. Will change.

File: loaders/chains.go (hunk 2 above)
Comment (VoR0220) [inline loaders/chains.go]: Separate PR

File: pkgs/jobs/jobs_transact.go (hunk 3 above)
Comment (VoR0220) [inline pkgs/jobs/jobs_transact.go]: Separate PR

File: loaders/chains.go (hunk 1 above)
Comment (therealplato) [inline loaders/chains.go]: `"Your chain types <chain type?> definition file confused the marmots, please check its formatting."`

Comment (VoR0220) [inline loaders/chains.go]: :) Excellent suggestion! I think I'll pop that in there in the next PR.

File: pkgs/jobs/writers.go (hunk 4 above)
Comment (zramsay) [inline pkgs/jobs/writers.go]: #1171
